bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/firmware/FirmwareStatusInfo.java

```diff
@@ -37,76 +45,104 @@ protected FirmwareStatusInfo() {
         // does nothing at all
     }
 
-    private FirmwareStatusInfo(FirmwareStatus firmwareStatus, FirmwareUID updatableFirmwareUID) {
+    private FirmwareStatusInfo(ThingUID thingUID, FirmwareStatus firmwareStatus, Firmware updatableFirmware) {
+        Objects.requireNonNull(thingUID, "Thing UID must not be null.");
+        this.thingUID = thingUID;
+
         Objects.requireNonNull(firmwareStatus, "Firmware status must not be null.");
         this.firmwareStatus = firmwareStatus;
-        this.updatableFirmwareUID = updatableFirmwareUID;
+
+        // Keep only the feasible information from the firmware to pass it through event later
+        if (updatableFirmware != null) {
+            this.thingTypeUID = updatableFirmware.getThingTypeUID();
```

Comment:
I wonder why we need the thing type uid here

Reply:
In general we can identify the thing type UID by the thing UID so I agree that we can remove the thing type uid from here.